codegen/gateway.go

```diff
@@ -564,12 +563,33 @@ func NewEndpointSpec(
 	return augmentHTTPEndpointSpec(espec, endpointConfigObj, midSpecs)
 }
 
+func reloadTestFixtures(espec *EndpointSpec, endpointConfigObj map[string]interface{}) (*EndpointSpec, error) {
+	field, ok := endpointConfigObj["testFixtures"]
+	if !ok {
+		return espec, nil
+	}
+	testFixturesRaw, err := json.Marshal(field)
+	if err != nil {
+		return espec, err
+	}
+	var ret map[string]*EndpointTestFixture
+	err = json.Unmarshal(testFixturesRaw, &ret)
+	if err != nil {
+		return espec, err
+	}
+	espec.TestFixtures = ret
+	return espec, nil
+}
+
 func augmentHTTPEndpointSpec(
 	espec *EndpointSpec,
 	endpointConfigObj map[string]interface{},
 	midSpecs map[string]*MiddlewareSpec,
 ) (*EndpointSpec, error) {
-	espec.TestFixtures = endpointConfigObj["testFixtures"].([]interface{})
+	espec, err := reloadTestFixtures(espec, endpointConfigObj)
+	if err != nil {
+		return nil, errors.Errorf("Unable to parse test cases")
```

Comment:
Don't drop the original error here. Use `errors.Wrap()`

Reply:
good point